Migrate to new survey presentation APIs

mmertsock opened the pull request:
## Overview

Implements #52. Replaces the "embeddable survey" feature implemented in #14.

The new survey APIs allow presenting any survey that's published to a given project, and simplify the process to present that survey. All surveys are launched using the survey name. No other identifiers are needed—particularly, participant link identifiers are no longer necessary. The actual user experience is basically the same as the embeddable survey feature: the app launches a view controller that displays the survey in a WKWebView. Example app and documentation updated.

## Security

REMINDER: All file contents are public.

- [X] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc. Of particular note:
    - No temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
    - Xcode project/target settings should remain generic. Don't leak team identifiers, code signing info, local filesystem paths, etc.
- [X] My changes do not introduce any security risks, or any such risks have been properly mitigated.

Describe briefly what security risks you considered, why they don't apply, or how they've been mitigated.

- Surveys are identified using surveyName, which is a well known identifier. The survey name is controlled/configured by the consumer of this SDK, and URLs are constructed using this survey name; all such usages of the survey name apply appropriate measures such as standard APIs to ensure proper query string encoding and avoid modification of the URL hostname/path/etc.
- The survey is launched in a WKWebView, which displays a hard-coded HTML document that embeds the survey itself via an iframe. Data is passed from the SDK native code to the web content by embedding a JSON object into the HTML string. JSONEncoder is used to ensure that all data is properly escaped and includes no executable code.
- Authentication is based on the existing access token functionality of `ParticipantSession`; the token is embedded into the wrapper HTML string so that the embedded survey can use the token.

## Checklist

- [X] All public symbols are documented using Swift Markup comments.
- [ ] If this feature requires a developer doc update, tag `@CareEvolution/api-docs`.
- [X] Source code file header comments are clean and standardized. Use "Created by CareEvolution on m/dd/yy".
- [X] Test and update the example app as needed. The example app should demonstrate all features of the SDK, and it's the most convenient way to test your feature.

Consider "Squash and merge" as needed to keep the commit history reasonable on `main`.

![Simulator Screen Recording - iPhone 14 Pro - 2023-03-13 at 14 43 30](https://user-images.githubusercontent.com/2637734/224799541-4a7f90fb-f406-45bf-a05b-30a41cc63eb3.gif)

mmertsock committed:
Standalone survey implementation

mmertsock committed:
SurveyLauncherView

mmertsock committed:
Documentation

mmertsock committed:
Remove obsolete embeddable survey feature

mmertsock committed:
Prevent swipe dismissal

mmertsock committed:
More cleanup

mmertsock committed:
Support for saved survey result

mmertsock committed:
Improved docs: platform support, tasks

eschramm commented on MyDataHelpsKit/Model/SurveyPresentation.swift:
```diff
@@ -0,0 +1,140 @@
+//
+
+import Foundation
+import RegexBuilder
```
I didn't see any regex parsing. Is this still needed?

mmertsock replied:
Good catch

eschramm reviewed (approved):
one minor question, otherwise LGTM 👍

mmertsock committed:
Remove link identifiers

mmertsock committed:
Launch surveys from task view

mmertsock commented:
Pushed a few more changes:

- PR feedback
- Removed participant link identifiers and survey task link identifiers. These were only used for the old embedded survey implementation
- Added the ability to launch a survey from the survey task list view

eschramm commented:
Changes LGTM

mmertsock committed:
Enable CI tests on v2 branch
